prediction_market_agent_tooling/markets/omen/omen.py

```diff
@@ -581,17 +581,44 @@ def get_conditional_tokens_balance_for_market(
 
 def omen_remove_fund_market_tx(
     market: OmenAgentMarket,
-    shares: OmenOutcomeToken,
+    shares: Wei | None,
     from_private_key: PrivateKey,
-    auto_withdraw: bool,
+    web3: Web3 | None = None,
 ) -> None:
+    """
+    Removes funding from a given OmenMarket (moving the funds from the OmenMarket to the
+    ConditionalTokens contract), and finally calls the `mergePositions` method which transfers collateralToken from the ConditionalTokens contract to the address corresponding to `from_private_key`.
+    """
+    from_address = private_key_to_public_key(from_private_key)
     market_contract = market.get_contract()
-    market_contract.removeFunding(shares, from_private_key)
 
-    # TODO: How to withdraw remove funding back to our wallet.
-    # Then also add to the test in tests_integration/markets/omen/test_omen.py.
-    if auto_withdraw:
-        raise NotImplementedError("TODO")
+    total_shares = market_contract.balanceOf(from_address, web3=web3)
+    if total_shares == 0:
+        logger.info("No shares to remove. Exiting.")
+        return
+
+    if not shares or shares > total_shares:
```

Comment:
If I call this with shares=0, it will unfund everything, Could it be like this?

```suggestion
    if shares is None or shares > total_shares:
```

Reply:
Good catch. Implementing this.